JailMonkey/JailMonkey.m

```diff
@@ -20,6 +36,323 @@ @implementation JailMonkey
 
 RCT_EXPORT_MODULE();
 
+#define A(c)            (c) - 0x19
+#define HIDE_STR(str)   do { char *p = str;  while (*p) *p++ -= 0x19; } while (0)
+typedef int (*ptrace_ptr_t)(int _request, pid_t _pid, caddr_t _addr, int _data);
+#if !defined(PT_DENY_ATTACH)
+#define PT_DENY_ATTACH 31
+#endif
+
+BOOL DEBUGGING = YES;
+
+void LOG(NSString* loc)
+{
+    NSLog(@"Found: %@", loc);
+}
+
+// Preventing libobjc hooked, strstr implementation
+const char* tuyul(const char* X, const char* Y)
+{
+    if (*Y == '\0')
+        return X;
+
+    for (int i = 0; i < strlen(X); i++)
```

Comment:
this logic bit confuses for me, rarely see recurring function invoke with loop. 
Have you tested it, does is serve the purpose?

Reply:
It does. I've tested it. 

It's very elegant how it detects dynamic library injections when you think about it (and read the code). It is in a way strange code, but it does work like a charm. This function is the corner stone of the approach. 

Of cause I didn't come up with this. I lifted it from here : https://github.com/avltree9798/isJailbroken/blob/master/isJailbroken/JB.m#L74

Reply:
I think this function is supposed to match the substring  say,  X = "thisisasteak",  Y="steak".

but for this  X="AaBbCc" Y="abc"?    it is gonna return non-null for this as well.
I think We are looking for exact match, sorry I didn't test the func, may miss something for the code.